src/kvstore/LogEncoder.h

```diff
@@ -40,12 +47,49 @@ std::string encodeMultiValues(LogType type,
                               folly::StringPiece v2);
 std::vector<folly::StringPiece> decodeMultiValues(folly::StringPiece encoded);
 
+std::string encodeBatchValue(const std::vector<std::pair<BatchLogType,
+                             std::pair<std::string, std::string>>>& batch);
+
+std::vector<std::pair<BatchLogType, std::pair<std::string, std::string>>>
+decodeBatchValue(folly::StringPiece encoded);
 
 std::string encodeHost(LogType type, const HostAddr& learner);
 HostAddr decodeHost(LogType type, const folly::StringPiece& encoded);
 
 int64_t getTimestamp(const folly::StringPiece& command);
 
+
+class BatchLogHelper {
```

Comment:
Generally LGTM 🎉  Personally think `BatchHolder` would be better？

Reply:
Good idea, let me rename it.